### ISSUE TITLE
Add Stage Target

### ISSUE THREAD

**Issue by cmmorrow** [cmmorrow/build-magic#1]
**Is your feature request related to a problem? Please describe.**
Currently, all Stages in a config file must be executed to pass and are executed in the order they are defined.  However, there might be instances where a user wants to run a particular stage or stages in a different order.

**Describe the solution you'd like**
The "target" feature can be used similar to the `make` command, where targets for make to execute are provided as arguments. In this case, the `--target` option can be added to the CLI, where the specified value matches a corresponding Stage name. Multiple `--target` options can be provided to run multiple Stages, and the Stages should be executed in the order they're provided.

To summarize:
* Add the --target argument to execute an individual stage in a config file.
* Allow for multiple --target args and execute them in order.
* Allow for a "default" config file name of **build-magic.yaml** so that the --config option can be omitted.
* Hardcode an "all" target which runs all the stages in order in the config file.
* Contextually parse a single arg as a target stage name as syntactic sugar so that the --target option can be eliminated.
